components/Button.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+interface ButtonProps {
+  label?: string;
+  children?: React.ReactNode;
+  onClick?: () => void;
+  disabled?: boolean;
+  color?: string;
+  backgroundColor?: string;
+  borderRadius?: string;
+  href?: string;
+  className?: string;
+  hoverColor?: string;
+  padding?: [string, string];
+  margin?: [string, string];
+}
+
+const Button: React.FC<ButtonProps> = ({
+  label,
+  children,
+  onClick,
+  disabled = false,
+  color = 'text-white',
+  backgroundColor = 'bg-color-1',
+  borderRadius = 'rounded-md',
+  href,
+  className,
+  hoverColor = 'hover:bg-hover-1',
+  padding = ['px-0', 'py-0'],
+  margin = ['mx-0', 'my-0'],
+}) => {
+  // Runtime validation to ensure only label or children is provided
+  if ((label && children) || (!label && !children)) {
```

Comment:
I think here we could make two interfaces something like:
```ts
interface Props {
  disabled: boolean
}
interface LabelProps extends Props {
  label: string;
}
interface ChildrenProps {
  children: React.ReactNode;
}
const button = (props: LabelProps | ChildrenProps) => {
  // Perform type deconstruction
  if (props.children != null) // This will probably need to be like hasownProperty or something.
}
```

its a bit overkill so if you want todo it at runtime im okay with that (we are the only devs, and will notice the runtime error for sure). Just a future note you can apply this pattern here if you want though.


Also do we want to combine this with the `icon` button, just handle it all in one its probably best if you either do the full type compile type typechecking in this case, or alternatively we could allow for an icon and a label or an icon and children the icon goes on the left and the content on the right. if there is just an icon we just center it.

Reply:
Fixed